imap_processing/hi/l1c/hi_l1c.py

```diff
@@ -115,6 +115,20 @@ def allocate_pset_dataset(n_esa_steps: int, sensor_str: str) -> xr.Dataset:
         dims=["esa_energy_step"],
         attrs=attrs,
     )
+    # TODO: define calibration product number to coincidence type mapping and
+    #     use the number of calibration products here. I believe it will be 5
+    #     0 for any, 1-4, for the number of detector hits.
+    n_calibration_prod = 5
+    attrs = attr_mgr.get_variable_attributes(
```

Comment:
Why is check_schema=False? I can't remember.

Reply:
SAMMI will strip off all of the `dtype` entries if I don't pass `check_schema=False`.